src/internal/connector/data_collections.go

```diff
@@ -37,6 +37,7 @@ func (gc *GraphConnector) ProduceBackupCollections(
 	owner common.IDNamer,
 	sels selectors.Selector,
 	metadata []data.RestoreCollection,
+	priorVersion int,
```

Comment:
nit: maybe either a comment on what this is or rename to `minMetadataVersion`? Longer term this does open up the question of if the version should be associated with the collections themselves as now that you've written the description it's possible we may treat different versions differently

Reply:
Going to rename this to `priorBackupVersion`, since it isn't specific for tracking metadata.